action/protocol/staking/handlers.go

```diff
@@ -314,6 +315,9 @@ func (p *Protocol) handleChangeCandidate(ctx context.Context, act *action.Change
 	if rErr = validateBucketSelfStake(featureCtx, csm, bucket, false); rErr != nil {
 		return log, rErr
 	}
+	if rErr := validateBucketEndorsement(NewEndorsementStateManager(csm.SM()), bucket, false, blkCtx.BlockHeight); rErr != nil {
```

Comment:
1. should this be protected by hard-fork flag?
2. if bucket status = `UnEndorsing`, it will return `nil`.  This case is considered OK to change candidate?

Reply:
1. I think not required, b/c there should be no endorsement before HF, so the validation should pass
2. It actually returns error if bucket is UnEndorsing